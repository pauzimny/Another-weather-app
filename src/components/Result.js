import React from "react";
import { Link } from "react-router-dom";
import Button from "@material-ui/core/Button";
import DeleteIcon from "@material-ui/icons/Delete";
import Card from "@material-ui/core/Card";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(theme => ({
  card: {
    display: "flex",
    width: "100%",
    margin: "20px auto 20px -5%",
    padding: "0 5% 0 5%"
  },
  deleteBtn: {
    backgroundColor: "red",
    color: "#fff",
    "&:hover": {
      backgroundColor: "#c51162"
    }
  },
  cityButton: {
    margin: theme.spacing(1),
    padding: "2px"
  }
}));

const Result = props => {
  const classes = useStyles();
  return props.city ? (
    <Card className={classes.card}>
      <li className="result">
        <Link
          className="result__city"
          to={{
            pathname: `/details/${props.city}`
          }}
        >
          <Button color="secondary" className={classes.cityButton}>
            {props.city}
          </Button>
        </Link>
        <span className="result__temp">
          {props.temp} {props.units !== "metric" ? "F" : "\u00b0C"}
        </span>

        <Button
          className={classes.deleteBtn}
          variant="contained"
          size="medium"
          onClick={() => props.delete(props.id)}
        >
          Usu≈Ñ
          <DeleteIcon />
        </Button>
      </li>
    </Card>
  ) : null;
};

export default Result;
