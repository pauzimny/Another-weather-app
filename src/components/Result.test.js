import React from "react";
import Result from "./Result.js";
import { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";

it("renders correctly", () => {
  const wrapper = shallow(
    <Result
      city={"london"}
      temp={18}
      units={"metric"}
      onClick={() => {}}
      id={2}
    />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
