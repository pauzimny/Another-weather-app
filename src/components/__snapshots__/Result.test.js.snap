// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <li
    className="result"
  >
    <Link
      className="result__city"
      to={
        Object {
          "pathname": "/details/london",
        }
      }
    >
      <span>
        london
      </span>
    </Link>
    <span
      className="result__temp"
    >
      18
       
      °C
    </span>
    <button
      className="result__delete"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="minus-circle"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        style={
          Object {
            "paddingRight": "5px",
          }
        }
        symbol={false}
        title=""
        transform={null}
      />
      Usuń
    </button>
  </li>
</Fragment>
`;
